internal/operands/vm-console-proxy/reconcile.go

```diff
@@ -250,14 +248,7 @@ func reconcileRoute(serviceName string) common.ReconcileFunc {
 }
 
 func getVmConsoleProxyNamespace(request *common.Request) string {
-	const defaultNamespace = "kubevirt"
-	if request.Instance.GetAnnotations() == nil {
-		return defaultNamespace
-	}
-	if namespace, isFound := request.Instance.GetAnnotations()[VmConsoleProxyNamespaceAnnotation]; isFound {
-		return namespace
-	}
-	return defaultNamespace
+	return "kubevirt"
```

Comment:
@akrejcir As you've mentioned, `kubevirt` namespace is no longer needed to be default one (and IIUC, no namespace is needed at all to be provided by the user). That being said, `getVmConsoleProxyNamespace` can be refactored to use SSP's CR namespace.

So just to confirm, in https://github.com/kubevirt/ssp-operator/pull/609#discussion_r1246259536 you've mentioned that we can deploy vm-console-proxy resources in SSP's CR namespace, right?

/cc @lyarwood @0xFelix

Reply:
How is the deployment of the template validator handled? Can we do the same for vm-console-proxy?

Reply:
Okay, but I see the implementation of `NamespacedResource()` that is:
```
func (r *reconcileBuilder) NamespacedResource(resource client.Object) ReconcileBuilder {
	r.resource = resource
	r.isClusterResource = false
	return r
}
```
I don't see in the code how it would set the object's namespace, that was done by `someObject.namespace = someNamespace`. Do you know by chance? maybe we need to anyway keep setting the namespace? e.g. `serviceAccount.namespace = request.Instance.Namespace`?